dali/operators/generic/slice/slice_base.cc

```diff
@@ -74,20 +74,33 @@ template <typename OutputType, typename InputType, int Dims>
 void SliceBaseCpu<OutputType, InputType, Dims>::RunImpl(workspace_t<CPUBackend> &ws) {
   const auto &input = ws.template InputRef<CPUBackend>(0);
   auto &output = ws.template OutputRef<CPUBackend>(0);
+  output.SetLayout(input.GetLayout());
+
   int nsamples = input.size();
   auto& thread_pool = ws.GetThreadPool();
   auto out_shape = output.shape();
-  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
-    thread_pool.AddWork(
-      [this, &input, &output, sample_idx](int thread_id) {
-        auto in_view = view<const InputType, Dims>(input[sample_idx]);
-        auto out_view = view<OutputType, Dims>(output[sample_idx]);
-        kernels::KernelContext ctx;
-        kmgr_.Run<Kernel>(thread_id, sample_idx, ctx, out_view, in_view, args_[sample_idx]);
-      }, out_shape.tensor_size(sample_idx));
+
+  auto in_view = view<const InputType, Dims>(input);
+  auto out_view = view<OutputType, Dims>(output);
+
+  if (nsamples < thread_pool.NumThreads() * 4) {  // intra-sample parallelization
```

Comment:
Maybe it it should compare size of the input. In this case when there is a great size inequality in sample sizes it can be suboptimal.